.github/workflows/check_new_kernel_configs.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+
+# find the lines in the diff that contain +/- CONFIG
+# ignore the lines that contain @
+def find_matching_lines(input_string):
+    pattern = r'(?!.*@)[+-]\s*.*CONFIG.*'
+    matching_lines = re.findall(pattern, input_string)
+
+    return matching_lines
+
+# parse diff for new kernel configs
+# check if they are in required configs
+def find_missing_words(json_file, string):
+    # Load the JSON object
+    with open(json_file, 'r') as file:
+        data = json.load(file)
+    configData = data['required-configs']
+
+    # Extract the words from the string
+    config_words = find_matching_lines(string)
+    holder = [ s.replace('+', '').replace('-', '').replace('=y','').replace('=m','').replace(' is not set', '').replace('#', '').strip() for s in config_words]
+    wordSet = set(holder)
+
+    # Find the missing words
+    missing_words = []
+    for word in wordSet:
+        if word not in configData:
+            missing_words.append(word)
```

Comment:
I'm curious if using a [set difference](https://www.geeksforgeeks.org/python-set-difference/) would work.

Reply:
had to expand the check to account for the arch (config could be in AMD but not ARM and thus is still missing)